src/Contract/Config/Collector/ComposerConfig.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace Qossmic\Deptrac\Contract\Config\Collector;
+
+use Qossmic\Deptrac\Contract\Config\CollectorConfig;
+use Qossmic\Deptrac\Contract\Config\CollectorType;
+
+final class ComposerConfig extends CollectorConfig
+{
+    protected CollectorType $collectorType = CollectorType::TYPE_COMPOSER;
+
+    /** @var list<string> */
+    private array $packages = [];
+
+    private function __construct(
+        private readonly string $composerPath,
+        private readonly string $composerLockPath,
+    ) {
+    }
+
+    /**
+     * @param list<string> $packages
+     */
+    public static function create(string $composerPath, string $composerLockPath, array $packages = []): self
```

Comment:
Would it make sense to make `$composerPath` and `$composerLockPath` optional and set `composer.json` and `composer.lock` as default values? That might reduce some unnecessary redundancy, considering most people will use these anyway.

Reply:
Sure.